### PR TITLE
view_as_real doesn't work for all backends since it relies on strides.

### PR DESCRIPTION
**Pull request by ailzhang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47018 view_as_real doesn't work for all backends since it relies on strides.**

Differential Revision: [D24607340](https://our.internmc.facebook.com/intern/diff/D24607340)

### CODE REVIEW

**Review by anjali411 — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -283,7 +283,7 @@
   use_c10_dispatcher: full
   variants: function
   dispatch:
-    DefaultBackend: view_as_real
+    CPU, CUDA: view_as_real
```
Comment (albanD) [inline aten/src/ATen/native/native_functions.yaml:286]: I guess the same is true for view_as_complex just below?

Comment (ailzhang) [reply aten/src/ATen/native/native_functions.yaml:286]: ah nice catch!

**Review by albanD — APPROVED** [review]
LGTM thanks for the fix!

**Comment by anjali411** [comment]
should similar changes be made for other view functions as well? For example, `torch.t()` (which currently dispatches to `DefaultBackend`)

**Comment by albanD** [comment]
I would say yes.
Especially because we can easily test this behavior by trying to call this function with a sparse Tensor.

**Pull request closed without merging**